collect_app/src/main/java/org/odk/collect/android/tasks/ServerPollingJob.java

```diff
@@ -0,0 +1,204 @@
+            }
+
+            new JobRequest.Builder(TAG)
+                    .setPeriodic(period, 300000)
+                    .setUpdateCurrent(true)
+                    .build()
+                    .schedule();
+        }
+    }
+
+    private boolean wasThisNewerFormVersionAlreadyDetected(String formVersionHash) {
+        Cursor cursor = new FormsDao().getFormsCursor(LAST_DETECTED_FORM_VERSION_HASH + "=?", new String[]{formVersionHash});
+        return cursor == null || cursor.getCount() > 0;
+    }
+
+    private void informAboutNewAvailableForms() {
+        Intent intent = new Intent(getContext(), FormDownloadList.class);
+        intent.putExtra(DISPLAY_ONLY_UPDATED_FORMS, true);
+        PendingIntent contentIntent = PendingIntent.getActivity(getContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(getContext())
+                .setSmallIcon(R.drawable.notes)
+                .setContentTitle(getContext().getString(R.string.form_updates_available))
+                .setAutoCancel(true)
+                .setContentIntent(contentIntent);
+
+        NotificationManager manager = (NotificationManager) getContext().getSystemService(Context.NOTIFICATION_SERVICE);
+        if (manager != null) {
+            manager.notify(0, builder.build());
+        }
+    }
+
+    private void informAboutNewDownloadedForms(String message) {
+        Intent intent = new Intent(Collect.getInstance(), NotificationActivity.class);
+        intent.putExtra(NotificationActivity.NOTIFICATION_KEY, message);
+        PendingIntent contentIntent = PendingIntent.getActivity(getContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(getContext())
+                .setSmallIcon(R.drawable.notes)
+                .setContentTitle(getContext().getString(R.string.new_form_versions_downloaded))
+                .setAutoCancel(true)
+                .setContentIntent(contentIntent);
+
+        NotificationManager manager = (NotificationManager) getContext().getSystemService(Context.NOTIFICATION_SERVICE);
+        if (manager != null) {
+            manager.notify(0, builder.build());
+        }
+    }
+
+    private void updateLastDetectedFormVersionHash(String formId, String formVersionHash) {
+        ContentValues values = new ContentValues();
+        values.put(LAST_DETECTED_FORM_VERSION_HASH, formVersionHash);
+        new FormsDao().updateForm(values, JR_FORM_ID + "=?", new String[] {formId});
+    }
+
+    private boolean isDeviceOnline() {
+        ConnectivityManager connMgr =
+                (ConnectivityManager) Collect.getInstance().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
+        return (networkInfo != null && networkInfo.isConnected());
+    }
+
+    public static void pollServerIfNeeded() {
```

Comment:
Similar comment to above, this method *could* be extracted out to another class.